### PR TITLE
HTMLFormatBear.py: Add HtmlFormatBear

### PR DESCRIPTION
**Pull request by Kiy4h**
Closes https://github.com/coala/coala-bears/issues/1869

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: bears/hypertext/HTMLFormatBear.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import json
+
+from typing import List, Union
```
Comment (ayan-b) [inline bears/hypertext/HTMLFormatBear.py]: This is an in-built dependency, so it should be with import json.

File: bears/hypertext/HTMLFormatBear.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import json
+from typing import List, Union
+from coalib.bearlib.abstractions.Linter import linter
```
Comment (ayan-b) [inline bears/hypertext/HTMLFormatBear.py]: A new line is missing here.

Comment (ayan-b) [inline bears/hypertext/HTMLFormatBear.py]: A newline before this line is missing.

File: tests/hypertext/HTMLFormatBearTest.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+<head>
+  <title>Foo</title>
+</head>
+<body>
+  <p>Bar</p>
+</body>
+</html>""".split('\n')
+
+
+bad_file_1 = """<!doCTYPE HTML><html>
```
Comment (bkhanale) [inline tests/hypertext/HTMLFormatBearTest.py]: Test the bad_files with their output

File: bears/hypertext/HTMLFormatBear.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        :param indent:
+            Indentation per level. When number, uses that amount of spaces. When
+            string, uses that per indentation level.
+        :param indent_initial:
+            Whether to indent the first level.
+        :param blanks:
+            List of tag-names, which, when next to each other, are joined by a
+            blank line (\n\n).
+        """
+        rehype_format_configs = {
```
Comment (bkhanale) [inline bears/hypertext/HTMLFormatBear.py]: This isn't tested as well

File: bears/hypertext/HTMLFormatBear.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        output_format='corrected',
+        use_stdin=True)
+class HTMLFormatBear:
+    """
+    Formats HTML automatically using rehype-format.
+    """
+    LANGUAGES = {'HTML'}
+    REQUIREMENTS = {NpmRequirement('rehype-cli', '7.0.0'),
+                    NpmRequirement('rehype-format', '2.3.1'),
+                    }
```
Comment (li-boxuan) [inline bears/hypertext/HTMLFormatBear.py]: Can be changed to
```python
    REQUIREMENTS = {
        NpmRequirement('rehype-cli', '7.0.0'),
        NpmRequirement('rehype-format', '2.3.1'),
    }

**Pull request closed without merging**